packages/client/src/app/pages/portfolio/blogs/blogs.component.ts

```diff
@@ -0,0 +1,162 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { FormGroup, FormControl } from '@angular/forms';
+
+import { FrontendUser } from '@pulp-fiction/models/users';
+import { Blog, SetPublishStatus } from '@pulp-fiction/models/blogs';
+import { PaginateResult } from '@pulp-fiction/models/util';
+import { AuthService } from '../../../services/auth';
+import { Constants, Title } from '../../../shared';
+import { PortBlogs } from '../../../models/site';
+import { MatDialog } from '@angular/material/dialog';
+import { CreateBlogComponent, EditBlogComponent, PreviewBlogComponent } from '../../../components/modals/blogs';
+import { BlogsService } from '../../../services/content';
+
+@Component({
+    selector: 'port-blogs',
+    templateUrl: './blogs.component.html',
+    styleUrls: ['./blogs.component.less']
+})
+export class BlogsComponent implements OnInit {
+    currentUser: FrontendUser;
+    portUser: FrontendUser;
+    blogsData: PaginateResult<Blog>
+    userBlogsData: PaginateResult<Blog>
+
+    pageNum = 1;
+    listView = false;
```

Comment:
careful with this one. If this page ever has a third state, a boolean isn't going to cut it anymore.
(on the other hand, if it only ever has two states, booleans are way simpler)

Reply:
yeah, for now, there’s only ever going to be two states for these pages